### DISCUSSION TITLE
Error response format to front-end app

### DISCUSSION THREAD

**Discussion by metalalive** [Game-as-a-Service/Pandemic-Reign-of-Cthulhu#31] [General]
From [PR28](https://github.com/Game-as-a-Service/Pandemic-Reign-of-Cthulhu/pull/28)
We discuss the format and necessary bits for `GameError` class.
And current response body contains a field `reason` , which represents error code into one field.
```json
{"reason": 1004}
``` 

I forgot to mention that backend app does not need to merge the error code and function code into the same field , it should be OK to put them in separate fields in response body, such as :
```json
{"reason": 1004, "function": 1001}
``` 

Note that we can start another GitHub Issue to modify the response body in the API doc.

**Comment by teds-lin** [comment]
當後端API發生錯誤，前端會需要知道發生錯誤(http status code)，甚至更進一步會想要知道原因(`reason`)，但會需要暴露細節到function level給前端嗎？

**Reply by metalalive** [reply]
Let us wait for frontend people's reply, 
Currently this is not high-priority urgent task (that is my personal opinion).

when  frontend developers  build several user-interface visual components corresponding to backend response,
they should determine which data they require from the backend.

However it would be much better if frontned developers can consider this in advance.

**Comment by wingtkw** [comment]
In my opinion,
the front end should know the http status code & error message to know what went wrong
but I don't think we have to know the function level detail

**Comment by metalalive** [comment]
thanks for Wing's reply, close this discussion for now,
any maintainer can reopen it whenever it is necessary to modify the error response.
